File: workloads/model2.py
import sys

# flush=True is used to ensure that the output is sent immediately, which is important for inter-process communication.
print("READY", flush=True)


# variable initialized outside of logic, as Model 2 requires the workload to maintain state across invocations.
total = 0
for line in sys.stdin:
    n = int(line.strip())
    total += sum(i * i for i in range(n))
    print(total, flush=True)

File: workloads_rust/src/model2.rs
// Model 2 workload: stateful across invocations. Mirrors workloads/model2.py.
use std::io::{self, BufRead, Write};

fn compute(n: u64) -> u64 { (0..n).map(|i| i * i).sum() }

fn main() {
    println!("READY");
    io::stdout().flush().unwrap();
    let mut total: u64 = 0;
    for line in io::stdin().lock().lines() {
        let n: u64 = line.unwrap().trim().parse().unwrap();
        total += compute(n);
        println!("{}", total);
        io::stdout().flush().unwrap();
    }
}
